TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Qualifications.cshtml.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.SupportUi.Pages.Persons.PersonDetail;
+
+public class QualificationsModel : PageModel
+{
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+    private readonly ReferenceDataCache _referenceDataCache;
+
+    public QualificationsModel(
+        ICrmQueryDispatcher crmQueryDispatcher,
+        ReferenceDataCache referenceDataCache)
+    {
+        _crmQueryDispatcher = crmQueryDispatcher;
+        _referenceDataCache = referenceDataCache;
+    }
+
+    [FromRoute]
+    public Guid PersonId { get; set; }
+
+    [FromQuery]
+    public string? Search { get; set; }
+
+    [FromQuery]
+    public int? PageNumber { get; set; }
+
+    [FromQuery]
+    public ContactSearchSortByOption? SortBy { get; set; }
+
+    public string? Name { get; set; }
+
+    public MandatoryQualificationInfo[]? MandatoryQualifications { get; set; }
+
+    public async Task<IActionResult> OnGet()
+    {
+        var contactDetail = await _crmQueryDispatcher.ExecuteQuery(
+            new GetContactDetailByIdQuery(
+                PersonId,
+                new ColumnSet(
+                    Contact.Fields.FirstName,
+                    Contact.Fields.MiddleName,
+                    Contact.Fields.LastName,
+                    Contact.Fields.dfeta_StatedFirstName,
+                    Contact.Fields.dfeta_StatedMiddleName,
+                    Contact.Fields.dfeta_StatedLastName)));
+
+        Name = contactDetail!.Contact.ResolveFullName(includeMiddleName: false);
+
+        var qualifications = await _crmQueryDispatcher.ExecuteQuery(new GetQualificationsByContactIdQuery(PersonId));
+        MandatoryQualifications = await MapMandatoryQualifications(qualifications!);
+
+        return Page();
+    }
+
+    private async Task<MandatoryQualificationInfo[]> MapMandatoryQualifications(dfeta_qualification[] qualifications)
+    {
+        var mandatoryQualifications = new List<MandatoryQualificationInfo>();
+        foreach (var qualification in qualifications.Where(q => q.dfeta_Type == dfeta_qualification_dfeta_Type.MandatoryQualification).OrderByDescending(q => q.dfeta_CompletionorAwardDate))
```

Comment:
Could use LINQ here and avoid the `List<>` and `foreach`

Reply:
Done